druid/src/widget/textbox.rs

```diff
@@ -434,14 +434,13 @@ impl Widget<String> for TextBox {
 
     fn paint(&mut self, ctx: &mut PaintCtx, data: &String, env: &Env) {
         // Guard against changes in data following `event`
-        let content = if data.is_empty() {
-            &self.placeholder
+        if data.is_empty() {
+            self.selection = Selection::caret(0);
+            self.hscroll_offset = 0.;
```

Comment:
I don't think `paint` is the correct place for us to be modifying this state; we should just not allow it in the first place.

I would look into the events that modify selection, and try to make it so that we always keep our selection at 0 if we don't have real text?

Reply:
My bad.

The latest changes guard against`selection` being changed when receiving `EditAction::Click` and `EditAction::Drag` events on the empty text.